src/plugins/data/common/expressions/interpreter_provider.ts

```diff
@@ -71,8 +71,16 @@ export function interpreterProvider(config: any) {
       // if something failed, just return the failure
       if (getType(newContext) === 'error') return newContext;
 
+      // if execution was aborted return error
+      if (handlers.abortSignal && handlers.abortSignal.aborted) {
+        return createError({
```

Comment:
Moved this back to being `createError` because of the comment below:
> The interpreter should *never* fail. It should always return a `{type: error}` on failure

The consumers can check if `err.name === 'AbortError'` and handle those failures differently than other failures.